### PR TITLE
NSX-20: full test coverage for security_{tag,group,policy}

### PR DESCRIPTION
**Pull request by 0lvin**

### CODE REVIEW

File: cloudify_nsx/security/group.py [hunk 1]
```diff
@@ -65,6 +65,10 @@ def create(**kwargs):
             raise cfy_exc.NonRecoverableError(
                 "Security group '%s' already exists" % group['name']
             )
+        elif use_existing:
+            raise cfy_exc.NonRecoverableError(
+                "Security group '%s' does not exists" % group['name']
```
Comment (geokala) [inline cloudify_nsx/security/group.py]: Does not exist (rather than exists) (there are more of these below, but I won't comment all of them)

Comment (geokala) [inline cloudify_nsx/security/group.py]: (and yet "Security group <x> exists" is correct, so don't update that- English is a weird language...)

File: tests/platformtests/test_base.py [hunk 2]
```diff
@@ -0,0 +1,89 @@
+########
```
Comment (geokala) [inline tests/platformtests/test_base.py]: I wouldn't worry about generalising this too much more- we've got a new plugin tests framework we'll be shoving on the plugins in the hopefully near future.

File: cloudify_nsx/library/nsx_common.py [hunk 3]
```diff
@@ -276,6 +276,22 @@ def check_raw_result(result_raw):
         )
 
 
+def set_boolean_property(obj, path_list, value):
```
Comment (0lvin) [inline cloudify_nsx/library/nsx_common.py]: It has two goals:
* replace all:
-    if enabled:
 -        routing['routing']['enabled'] = "true"
 -    else:
 -        routing['routing']['enabled'] = "true"
To something less error prone.
* And receive "status" - do we really change something and need to update, or all values already as we want,

**Pull request merged**